### PR TITLE
Add 'lan' option (modify the option name to ‘useLocalIp’ for more semantic)

### PR DESCRIPTION
**Pull request by sliwey**
Let the browser open with local IP.

**What kind of change does this PR introduce?**
feature

**Did you add or update the `examples/`?**
no

**Summary**
According to [#147](https://github.com/webpack/webpack-dev-server/issues/147), server can be accessed via IP, but when set `host` to `0.0.0.0` and `open` to `true`, the browser will open  `http://0.0.0.0`, but it's inaccessible.

This PR allows to set ~~`lan`~~ `useLocalIp` to `true`, so that the browser can open with local IP.

**Example usage in webpack config:**
```javascript
devServer: {
  host: '0.0.0.0',
  disableHostCheck: true,
  open: true,
  // lan: true,  // set true to let the browser open with local IP
  useLocalIp: true,  // set true to let the browser open with local IP
  ...
}
```

**Does this PR introduce a breaking change?**
no

### CODE REVIEW

File: lib/util/createDomain.js [hunk 1]
```diff
@@ -1,13 +1,31 @@
 "use strict";
 const url = require("url");
+const networkInterfaces = require("os").networkInterfaces();
+
+function getLocalIP() {
```
Comment (bebraw) [inline lib/util/createDomain.js]: Can you write tests for this?

Comment (sliwey) [inline lib/util/createDomain.js]: No problem, I will add some tests later.

File: lib/util/getLocalIP.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+"use strict";
```
Comment (shellscape) [inline lib/util/getLocalIP.js]: Please use https://www.npmjs.com/package/internal-ip (first choice) or https://www.npmjs.com/package/dev-ip to replace the need for this file.

File: lib/util/createDomain.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+const getLocalIP = require("./getLocalIP");
 
 module.exports = function createDomain(options) {
 	const protocol = options.https ? "https" : "http";
 
 	// the formatted domain (url without path) of the webpack server
 	return options.public ? `${protocol}://${options.public}` : url.format({
 		protocol: protocol,
-		hostname: options.host,
+		hostname: options.lan ? getLocalIP() : options.host,
```
Comment (shellscape) [inline lib/util/createDomain.js]: I don't feel that `lan` is semantic for this feature. `options.useLocalIp` or some such would probably be easier to understand at first glance.

Comment (bebraw) [inline lib/util/createDomain.js]: Yeah, agreed. 👍

File: lib/util/getLocalIP.js (hunk 2 above)
Comment (sliwey) [inline lib/util/getLocalIP.js]: Good idea, I will use `internal-ip` to replace this file.

File: lib/util/createDomain.js (hunk 3 above)
Comment (sliwey) [inline lib/util/createDomain.js]: @shellscape  You're right, I will change the word to `useLocalIp` to make it more semantic.
And thank you for giving me all the advice.

**Comment by shellscape** [comment]
Looks good. Please sign the CLA (https://github.com/webpack/webpack-dev-server/pull/901#issuecomment-298868768) and we can get this merged

**Comment by sliwey** [comment]
@shellscape  I have already signed the CLA.

**Pull request merged**